contracts/examples/BalanceCredentialIssuer.sol

```diff
@@ -1,85 +1,106 @@
+import {NonMerklizedIssuerBase} from '@iden3/contracts/lib/NonMerklizedIssuerBase.sol';
+import {IdentityBase} from '@iden3/contracts/lib/IdentityBase.sol';
+import {PrimitiveTypeUtils} from '@iden3/contracts/lib/PrimitiveTypeUtils.sol';
+import {PoseidonUnit4L} from '@iden3/contracts/lib/Poseidon.sol';
 
 /**
  * @dev Example of decentralized balance credential issuer.
  * This issuer issue non-merklized credentials decentralized.
  */
-contract BalanceCredentialIssuer is IdentityBase, OwnableUpgradeable {
+contract BalanceCredentialIssuer is IdentityBase, NonMerklizedIssuerBase, OwnableUpgradeable {
     using IdentityLib for IdentityLib.Data;
 
-    // Information about the schema that is used
-    // for building verifiable credential from claim
-    struct SchemaInfo {
-        string jsonldSchemaURL;
-        string jsonSchemaURL;
-        string credentialType;
+    /// @custom:storage-location erc7201:balance.credential.issuer.storage
+    struct Storage {
+        // countOfIssuedClaims count of issued claims for incrementing id and revocation nonce for new claims
+        uint64 countOfIssuedClaims;
+        // claim store
+        mapping(uint256 => uint256[]) userClaims;
+        mapping(uint256 => ClaimItem) idToClaim;
+        // this mapping is used to store credential subject fields
+        // to escape additional copy in issueCredential function
+        // since "Copying of type struct OnchainNonMerklizedIdentityBase.SubjectField memory[] memory to storage not yet supported."
+        mapping(uint256 => INonMerklizedIssuer.SubjectField[]) idToCredentialSubject;
     }
 
-    // Claim representation
-    struct Claim {
-        uint256[8] claim;
+    bytes32 private constant STORAGE_LOCATION =
+        0x019dfb46485f789e684dc5f54f13de3e6ef161b4d8709723f90193e84ea09c53;
+
+    function getStorage() private pure returns (Storage storage $) {
+        bytes32 location = STORAGE_LOCATION;
+        assembly {
+            $.slot := location
+        }
     }
 
     /**
      * @dev Version of contract
      */
-    string public constant VERSION = "1.0.0";
+    string public constant VERSION = '1.0.0';
 
     // jsonldSchemaHash hash of jsonld schema.
     // More about schema: https://devs.polygonid.com/docs/issuer-node/issuer-node-api/claim/apis/#get-claims
-    uint256 private constant jsonldSchemaHash = 134825296953649542485291823871789853562;
+    uint256 private constant jsonldSchemaHash = 148834697620350657501993499321116864501;
+    string private constant jsonSchema = "https://gist.githubusercontent.com/ilya-korotya/e10cd79a8cc26ab6e40400a11838617e/raw/575edc33d485e2a4c806baad97e21117f3c90a9f/non-merklized-non-zero-balance.json";
+    string private constant jsonldSchema = "https://gist.githubusercontent.com/ilya-korotya/660496c859f8d31a7d2a92ca5e970967/raw/6b5fc14fe630c17bfa52e05e08fdc8394c5ea0ce/non-merklized-non-zero-balance.jsonld";
```

Comment:
Can we have balance as a big unsigned int instead of xsd:positiveInteger? So that it would be possible to set 0 to balance.